sds_data_manager/stacks/efs_lambda_stack.py

```diff
@@ -0,0 +1,147 @@
+        )
+
+        # Create a security group that allows network access to mount the EFS
+        self.efs_list_runs_lambda_sg = ec2.SecurityGroup(
+            self,
+            "LambdaEFSSecurityGroup",
+            vpc=vpc,
+            description="Allow calls to internet",
+            allow_all_outbound=True,
+            security_group_name="LambdaEFSSecurityGroup",
+        )
+
+        # NOTE: Workaround to overcome EFS circular dependency when mounting
+        # a filesystem
+        # https://github.com/aws/aws-cdk/issues/18759
+        efs_mount_config = scope.node.get_context("efs_mount_config")
+        efs_access_point_id = Fn.import_value(efs_mount_config["efs_access_point_id"])
+
+        # This access point is used by other resources to read from EFS
+        lambda_efs_access = aws_lambda.FileSystem(
+            arn=f"arn:aws:elasticfilesystem:{self.region}:{self.account}:access-point/{efs_access_point_id}",
+            local_mount_path=efs_mount_config["mount_path"],
+        )
+
+        lambda_code = aws_lambda.Code.from_asset(
+            "sds_data_manager/lambda_code/efs_lambda"
+        )
+
+        boto3_layer_arn = (
+            f"arn:aws:lambda:{self.region}:770693421928:layer:Klayers-p311-boto3:2"
+        )
+        layers = [
+            aws_lambda.LayerVersion.from_layer_version_arn(
+                self, id="boto3dependencylayer", layer_version_arn=boto3_layer_arn
+            )
+        ]
+
+        self.efs_list_runs_lambda = aws_lambda.Function(
+            self,
+            "EFSWriteLambda",
+            function_name="efs-write-lambda",
+            runtime=aws_lambda.Runtime.PYTHON_3_11,
+            code=lambda_code,
+            handler="lambda_function.lambda_handler",
+            role=efs_lambda_role,
+            description="Lambda that write data to the EFS",
+            # Access to the EFS requires to be within the VPC
+            vpc=vpc,
+            # Mount EFS access point to /mnt/data within the lambda
+            filesystem=lambda_efs_access,
+            timeout=Duration.minutes(1),
+            # Allow access to the EFS over NFS port
+            security_groups=[self.efs_list_runs_lambda_sg],
+            layers=layers,
+            architecture=aws_lambda.Architecture.ARM_64,
+            environment={
+                "EFS_MOUNT_PATH": efs_mount_config["mount_path"],
+            },
+        )
+
+        # Trigger lambda on all s3 object creations through
+        # eventbridge
+
+        # Define an EventBridge rule
+        event_rule = events.Rule(
+            self,
+            "EfsWriteLambdaS3EventRule",
+            event_pattern=events.EventPattern(
+                source=["aws.s3"],
+                detail_type=["Object Created"],
+                detail={
+                    "bucket": {"name": [data_bucket.bucket_name]},
+                    "object": {
+                        "key": [
+                            {"prefix": "imap"},
```

Comment:
```suggestion
                            {"prefix": "imap/spice"},
```
Can we add another prefix to just the spice folder as well?

Reply:
We kind of brainstormed on this. Rather than setting up two EventBridges we talked about combining them. Maybe put as a TODO here for another PR. Anyways, attaching the picture I took for reference. 

![IMG-5080](https://github.com/IMAP-Science-Operations-Center/sds-data-manager/assets/46567335/37dcfc67-5493-412c-bb95-d9e1bc072a0e)

Reply:
btw, this prefix is related to filename. all kernel file's name starts with `imap`. I know we use `imap` as root folder of our data bucket. I can see the confusion here.

Reply:
Ahh, so I think you'd need to put `prefix: imap/spice/imap` in that case with the double imap for the nested filename?

Reply:
true. since our folder structure will looks like that. will do that.